test/test_mtl_ata_translation.cpp

```diff
@@ -311,6 +313,106 @@ TEST_CASE("ATA satisfiability of simple MTL formulas", "[translator]")
+
+		// After reading a, phi is no longer satisfiable.
+		auto runs = ata.make_symbol_transition({{}}, a_symbol);
+		// Thus, we should end up in the sink location.
+		CAPTURE(runs);
+		REQUIRE(runs.size() == 1);
+		REQUIRE(runs[0].size() == 1);
+		CHECK(runs[0][0].second == Configuration{{sink, 0}});
+
+		// We should be able to loop in the sink location.
+		runs = ata.make_symbol_transition(ata.make_time_transition(runs, 0), a_symbol);
+		CAPTURE(runs);
+		REQUIRE(runs.size() == 1);
+		REQUIRE(runs[0].size() == 3);
+		CHECK(runs[0][2].second == Configuration{{sink, 0}});
+	}
+
+	SECTION("Sink location for an until transition")
+	{
+		const MTLFormula phi = a.until(b);
+		const auto       ata = translate(phi, {AP("a"), AP("b"), AP("c")});
+		CAPTURE(ata);
+
+		// After reading a -> 0 -> c, phi is no longer satisfiable.
+		auto runs = ata.make_symbol_transition(
+		  ata.make_time_transition(ata.make_symbol_transition({{}}, a_symbol), 0), c_symbol);
+		CAPTURE(runs);
+		REQUIRE(runs.size() == 1);
+		REQUIRE(runs[0].size() == 3);
+		// Thus, we should end up in the sink location.
+		CHECK(runs[0][2].second == Configuration{{sink, 0}});
+	}
+
+	SECTION("Sink location for a dual until transition")
+	{
+		const MTLFormula phi = a.dual_until(b);
+		const auto       ata = translate(phi, {AP("a"), AP("b"), AP("c")});
+		CAPTURE(ata);
+
+		// After reading a -> 0 -> c, phi is no longer satisfiable.
+		auto runs = ata.make_symbol_transition(
+		  ata.make_time_transition(ata.make_symbol_transition({{}}, a_symbol), 0), c_symbol);
+		CAPTURE(runs);
+		REQUIRE(runs.size() == 1);
+		REQUIRE(runs[0].size() == 3);
+		// Thus, we should end up in the sink location.
+		CHECK(runs[0][2].second.find(State{sink, 0}) != runs[0][2].second.end());
+	}
+
+	SECTION("Sink location for an until transition with a duration")
+	{
+		const MTLFormula phi = a.until(b, TimeInterval{0, 1});
+		const auto       ata = translate(phi, {AP("a"), AP("b"), AP("c")});
+		CAPTURE(ata);
+
+		// After reading a -> 2 -> b, phi is no longer satisfiable.
+		auto runs = ata.make_symbol_transition(
+		  ata.make_time_transition(ata.make_symbol_transition({{}}, a_symbol), 2), b_symbol);
+		CAPTURE(runs);
+		REQUIRE(runs.size() == 1);
+		REQUIRE(runs[0].size() == 3);
+		// Thus, we should end up in the sink location.
+		CHECK(runs[0][2].second == Configuration{{sink, 2}});
+	}
+
+	SECTION("Sink location for a dual until transition with a duration")
+	{
+		const MTLFormula phi = a.dual_until(b, TimeInterval{0, 1});
+		const auto       ata = translate(phi, {AP("a"), AP("b"), AP("c")});
+		CAPTURE(ata);
+
+		// After reading a -> 0 -> b, phi is no longer satisfiable.
+		auto runs = ata.make_symbol_transition(
+		  ata.make_time_transition(ata.make_symbol_transition({{}}, a_symbol), 0), a_symbol);
+		CAPTURE(runs);
+		REQUIRE(runs.size() == 1);
+		REQUIRE(runs[0].size() == 3);
+		// Thus, we should end up in the sink location.
+		CHECK(runs[0][2].second == Configuration{{sink, 0}});
+	}
 }
```

Comment:
Should we also end up in a sink if there is a conjunction of distinct APs? What about unsatisfiable disjunctions?